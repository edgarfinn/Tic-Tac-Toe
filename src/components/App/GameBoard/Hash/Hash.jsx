import React, { PureComponent } from 'react'
import styled from 'styled-components'
import HashLine from './HashLine/Hash_line'

class Hash extends PureComponent {
  render () {
    const HashGrid = styled.div`
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    padding: 10px;
    box-sizing: border-box;
    z-index: 0;
    `

    return (
      <HashGrid className='hash-grid'>
        <HashLine linePosition='top' transition='1' />
        <HashLine linePosition='bottom' transition='2' />
        <HashLine linePosition='left' transition='3' />
        <HashLine linePosition='right' transition='4' />
      </HashGrid>
    )
  }
};

export default Hash
