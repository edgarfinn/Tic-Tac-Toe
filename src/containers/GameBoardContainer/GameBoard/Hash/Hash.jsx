import React from 'react';
import styled from 'styled-components';
import HashLine from './HashLine/Hash_line';

const Hash = () => {
  const HashGrid = styled.div`
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    padding: 10px;
    box-sizing: border-box;
  `;

  return (
    <HashGrid>
      <HashLine linePosition="top" transition="1" />
      <HashLine linePosition="bottom" transition="2" />
      <HashLine linePosition="left" transition="3" />
      <HashLine linePosition="right" transition="4" />
    </HashGrid>
  );
};

export default Hash;
