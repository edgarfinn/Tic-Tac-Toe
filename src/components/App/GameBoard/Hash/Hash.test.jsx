/* eslint-env jest */

import React from 'react'
import 'jest-styled-components'
import Adapter from 'enzyme-adapter-react-16'
import Enzyme, { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import Hash from './Hash'

Enzyme.configure({ adapter: new Adapter() })

const setup = () => {
  const wrapper = shallow(<Hash />)
  return { wrapper }
}

describe('<Hash />', () => {
  const { wrapper } = setup()
  it('renders correctly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
