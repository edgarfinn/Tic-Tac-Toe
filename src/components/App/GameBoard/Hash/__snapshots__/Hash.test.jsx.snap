// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hash /> renders correctly 1`] = `
<styled.div>
  <HashLine
    linePosition="top"
    transition="1"
  />
  <HashLine
    linePosition="bottom"
    transition="2"
  />
  <HashLine
    linePosition="left"
    transition="3"
  />
  <HashLine
    linePosition="right"
    transition="4"
  />
</styled.div>
`;
